### PR TITLE
Fix/osidb 2573 save cvss on affect change

### PR DESCRIPTION
**Pull request by superbuggy**
# [OSIDB-ID] [Title]

## Checklist:

- [x] Linting passed
- [x] Type checks passed
- [x] Tests suite passed
- [x] Commits consolidated
- [x] Changelog updated
- [x] Test cases added/updated
- [x] Jira ticket updated

## Summary:

[Replace with a brief summary of the changes introduced by this PR.]

## Changes:

[Replace with a detailed description of the changes made in this PR, including any new features, enhancements, bug fixes, or refactorings.]

## Considerations:

[Replace with any additional considerations, notes, or instructions for reviewers.]

### CODE REVIEW

File: src/services/AffectService.ts [hunk 1]
```diff
@@ -35,3 +35,49 @@ export async function deleteAffect(uuid: string) {
...
+
+
+// {
+// "comment": "string",
+// "cvss_version": "V2",
+// "issuer": "RH",
+// "vector": "string",
+// "embargoed": true
+//   "updated_dt": "2024-02-06T16:02:54.708Z"
+// }
```
Comment (C-Valen) [inline src/services/AffectService.ts:48]: What is the purpose for this commented objects?
(same for lines 66 - 73)

File: src/services/AffectService.ts [hunk 2]
```diff
@@ -35,3 +35,49 @@ export async function deleteAffect(uuid: string) {
...
+export async function putAffectCvssScore(
+  affectId: string,
+  cvssScoresId: string,
+  cvssScoreObject: unknown,
+) {
```
Comment (C-Valen) [inline src/services/AffectService.ts:53]: Why having `cvssScoreObject ` as unknown?
(same for line 74)

File: src/services/AffectService.ts (hunk 1 above)
Comment (superbuggy) [reply src/services/AffectService.ts:48]: It's there to document the type used in the request, however, declaring the type here like we do in other such `*Service.ts` files may be preferable. This relates to a larger discussion on Zod/TS/OpenAPI type management

File: src/services/AffectService.ts (hunk 2 above)
Comment (superbuggy) [reply src/services/AffectService.ts:53]: It's basically a placeholder type, but I can declare these as actual types using the 'schema' in the comments above.

Comment (C-Valen) [reply src/services/AffectService.ts:53]: No preference from me

**Review by C-Valen — APPROVED** [review]

**Review by d10n — APPROVED** [review]

**Pull request merged**